src/Core/Commander/Providers/WMS_Provider.js

```diff
@@ -30,71 +32,221 @@ define('Core/Commander/Providers/WMS_Provider', [
+                        "," +
+                        (coord.minCarto.longitude - Math.PI)* 180.0 / Math.PI +
+                        ","+
+                       coord.maxCarto.latitude* 180.0 / Math.PI +
+                       "," +
+                       (coord.maxCarto.longitude - Math.PI )*180.0 / Math.PI;
+
+            var urld = url.replace('%bbox',bbox.toString());
+
+            return urld;
 
-        /**
-         * Returns the url for a WMS query with the specified bounding box
-         * @param {BoundingBox} bbox: requested bounding box
-         * @returns {Object@call;create.url.url|String}
-         */
-        WMS_Provider.prototype.url = function(bbox) {
-            var url = this.baseUrl + "?LAYERS=" + this.layer + "&FORMAT=" + this.format +
-                "&SERVICE=WMS&VERSION=1.1.1" + "&REQUEST=GetMap&BBOX=" +
-                bbox.minCarto.longitude + "," + bbox.minCarto.latitude + "," +
-                bbox.maxCarto.longitude + "," + bbox.maxCarto.latitude +
-                "&WIDTH=" + this.width + "&HEIGHT=" + this.height + "&SRS=" + this.srs;
-            return url;
         };
 
+        WMS_Provider.prototype.preprocessDataLayer = function(layer){
+            if(!layer.name)
+                throw new Error('layerName is required.');
+
+            layer.format = defaultValue(layer.options.mimeType, "image/png"),
+            layer.crs = defaultValue(layer.projection, "EPSG:4326"),
+            layer.width = defaultValue(layer.heightMapWidth, 256),
+            layer.version = defaultValue(layer.version, "1.3.0"),
+            layer.styleName = defaultValue(layer.style, "normal"),
+            layer.transparent = defaultValue(layer.transparent, false),
+            layer.bbox = defaultValue(layer.bbox, [-180, -90, 180, 90]);
+            layer.customUrl = layer.url +
+                          '?SERVICE=WMS&REQUEST=GetMap&LAYERS=' + layer.name +
+                          '&VERSION=' + layer.version +
+                          '&STYLES=' + layer.styleName +
+                          '&FORMAT=' + layer.format +
+                          '&TRANSPARENT=' + layer.transparent +
+                          '&BBOX=%bbox'  +
+                          '&CRS=' + layer.crs +
+                          "&WIDTH=" + layer.width +
+                          "&HEIGHT=" + layer.width;
+        };
+
+        WMS_Provider.prototype.tileInsideLimit = function(tile,layer) {
+                var bbox = tile.bbox;
+                // shifting longitude because of issue #19
+                var west =  layer.bbox[0]*Math.PI/180.0 + Math.PI;
+                var east =  layer.bbox[2]*Math.PI/180.0 + Math.PI;
+                var bboxRegion = new BoundingBox(west, east, layer.bbox[1]*Math.PI/180.0, layer.bbox[3]*Math.PI/180.0, 0, 0, 0);
+            return bboxRegion.intersect(bbox);
+        };
+
+        WMS_Provider.prototype.getColorTextures = function(tile, layer) {
```

Comment:
for data selection in providers, i have created an issue #157 but we must do it right now in this PR? because we must do for all providers, not only wms_provider.